### PR TITLE
Added support for resiliency module with minimal manifest

### PR DESCRIPTION
**Pull request by harishp8889**
# Description
Added support for resiliency module with minimal manifest

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1449 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested in Kubernetes cluster

### CODE REVIEW

File: operatorconfig/driverconfig/common/k8s-1.31-values.yaml [hunk 1]
```diff
@@ -21,5 +21,5 @@ images:
   sdcmonitor: dellemc/sdc:4.5.2.1
   # "images.metadataretriever" defines the container images used for csi metadata retriever
   metadataretriever: dellemc/csi-metadata-retriever:v1.8.0
-  # "images.csiReverseProxy" defines the container images used for reverse-proxy
-  csiReverseProxy: dellemc/csipowermax-reverseproxy:v2.11.0
+  # "images.csiReverseProxy" defines the container images used for reverse-proxy by default
+  csiReverseProxy: dellemc/csipowermax-reverseproxy:v2.10.0
```
Comment (bpjain2004) [inline operatorconfig/driverconfig/common/k8s-1.31-values.yaml:25]: Why the reverse proxy version is downgraded? Is it intentional?

File: operatorconfig/moduleconfig/resiliency/v1.11.0/container-powermax-controller.yaml [hunk 2]
```diff
@@ -14,8 +14,20 @@
 #
 #
 name: podmon
-image: dellemc/podmon:v1.11.0
+image: dellemc/podmon:v1.10.0
```
Comment (bpjain2004) [inline operatorconfig/moduleconfig/resiliency/v1.11.0/container-powermax-controller.yaml:17]: Why the podman version is downgraded? Is it intentional?

File: pkg/drivers/powermax.go [hunk 3]
```diff
@@ -94,17 +94,23 @@ func PrecheckPowerMax(ctx context.Context, cr *csmv1.ContainerStorageModule, ope
 			Value: "/var/lib/kubelet",
 		})
 	}
+	version, err := utils.GetLatestVersion(string(csmv1.ReverseProxy), operatorConfig)
```
Comment (bpjain2004) [inline pkg/drivers/powermax.go:97]: If the default config version is cosidered as the latest one, then can the same be considered for a minimal sample file?

File: samples/minimal-samples/powermax.yaml [hunk 4]
```diff
@@ -7,3 +7,6 @@ spec:
   driver:
     csiDriverType: "powermax"
     configVersion: v2.12.0
+  modules:
+    - name: resiliency
+      enabled: true
```
Comment (bpjain2004) [inline samples/minimal-samples/powermax.yaml:12]: Do we want to enable the resiliency module by default? If yes, can this behavior be made default and avoid these params in the minimal sample?

File: operatorconfig/driverconfig/common/k8s-1.31-values.yaml (hunk 1 above)
Comment (harishp8889) [reply operatorconfig/driverconfig/common/k8s-1.31-values.yaml:25]: As of now `2.11.0` is not released if we maintain that in the yamls then the installation will fail. Once we release the latest version we need to bump up here

File: operatorconfig/moduleconfig/resiliency/v1.11.0/container-powermax-controller.yaml (hunk 2 above)
Comment (harishp8889) [reply operatorconfig/moduleconfig/resiliency/v1.11.0/container-powermax-controller.yaml:17]: same as above comment

File: samples/minimal-samples/powermax.yaml (hunk 4 above)
Comment (harishp8889) [reply samples/minimal-samples/powermax.yaml:12]: nope we will not enable by default, The user has to give it explicitly. Let me add that in the comment

File: pkg/drivers/powermax.go (hunk 3 above)
Comment (harishp8889) [reply pkg/drivers/powermax.go:97]: If the user does not mention any version then the latest version is picked up by default. This is to reduce number of lines in the yaml.

**Review by bpjain2004 — APPROVED** [review]

**Review by meggm — APPROVED** [review]

**Pull request merged**